app/src/main/java/com/ptit/base/BaseViewModel.kt

```diff
@@ -0,0 +1,9 @@
+package com.ptit.base
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+abstract class BaseViewModel: ViewModel() {
+
+    val messageErrorLiveData = MutableLiveData<String>()
```

Comment:
Dùng backing property để tránh vô tình sửa giá trị livedata
```suggestion
    val messageErrorLiveData: LiveData<String> get() = _messageErrorLiveData
    private val _messageErrorLiveData = MutableLiveData<String>()
```

Reply:
Bạn nói đúng đó nha =))